pkg/skaffold/runner/runner.go

```diff
@@ -218,14 +222,18 @@ func (r *SkaffoldRunner) Deploy(ctx context.Context, out io.Writer, artifacts []
 	if err != nil {
 		return err
 	}
-	return r.performStatusCheck(out)
+	return r.performStatusCheck(ctx, out)
 }
 
-func (r *SkaffoldRunner) performStatusCheck(out io.Writer) error {
+func (r *SkaffoldRunner) performStatusCheck(ctx context.Context, out io.Writer) error {
 	// Check if we need to perform deploy status
 	if r.runCtx.Opts.StatusCheck {
 		fmt.Fprintln(out, "Waiting for deployments to stabilize")
-		// TODO : Actually perform status check
+		err := statusCheck(ctx, r.defaultLabeller, r.runCtx)
+		if err != nil {
+			fmt.Fprintln(out, err.Error())
```

Comment:
will this error be printed twice if we `return err` again next?

Reply:
no, it does not. 
https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/deploy.go#L55

```
func doDeploy(ctx context.Context, out io.Writer) error {
	return withRunner(ctx, func(r runner.Runner, _ *latest.SkaffoldConfig) error {
		// If the BuildArtifacts contains an image in the preBuilt list,
		// use image from BuildArtifacts instead
		deployArtifacts := build.MergeWithPreviousBuilds(buildOutputFile.BuildArtifacts(), preBuiltImages.Artifacts())

		return r.DeployAndLog(ctx, out, deployArtifacts)
	})
}
```
And DeployAndLog, does print it out either
```
func (r *SkaffoldRunner) DeployAndLog(ctx context.Context, out io.Writer, artifacts []build.Artifact) error {
	if !r.runCtx.Opts.Tail {
		return r.Deploy(ctx, out, artifacts)
	}
	....
	if err := r.Deploy(ctx, out, artifacts); err != nil {
		return err
	}
```

I thought about printing the error in cmd/skaffold/app/cmd/deploy.go but its called a lot of places and did not want to repeat it everywhere.

Reply:
makes sense! i think this answers a comment I made on your other PR as well :sweat_smile: